mammals/Mammals.cs

```diff
@@ -1,20 +1,54 @@
-class Mammal : Animal, ISpeciesInfo
+abstract class Mammal :   Animal, IVocalize
 {
-    public bool IsWarmBlooded {get; set; } = true; //Mammals are warm blooded
+    //All animals are warm blooded
+    public bool IsWarmBlooded {get;set;} = true;
+    private double averageBodyTemperature;
 
-    public Mammal(string name, DateTime birthDate, double weightInKg,
-        HabitatTypes habitat, string? originRegion = null, bool warmBlooded = true)
-        : base(name, birthDate, weightInKg, habitat, originRegion)
+    //Fur type e.g. none for dolphins, long for alpaca/sheep, etc...
+    public string? FurType {get; set;} 
+    //Ear type e.g. big for elephant, pointy for lynx, etc..
+    public string? EarType {get; set;}
+    //Tail type e.g. no tail for baboons, short/bushy for some type of cats
+    public string? TailType{get; set;}
+
+    public double AverageBodyTemperature
+    {
+        get => averageBodyTemperature;
+        set
+        {
+            if(value<30||value>45)
+                throw new ArgumentOutOfRangeException("Temperature out of expected mammalian range.");
+            else
+                averageBodyTemperature=value;
+        }
+    }
+    //mammal without any arguments.
+    public Mammal() : base()
```

Comment:
once this class is abstract all constructor access modifiers could be set as protected